### PR TITLE
Initial version of audience extensions sample

### PR DESCRIPTION
**Pull request by floelhoeffel**

### CODE REVIEW

File: samples/optimizely-audiences/index.html [hunk 1]
```diff
@@ -0,0 +1,134 @@
+<!doctype html>
+<html lang="en">
+
+<head>
+  <meta charset="UTF-8" />
+  <title>Optimizely audiences</title>
+
+  <link rel="stylesheet" href="https://contentful.github.io/ui-extensions-sdk/cf-extension.css">
```
Comment (jelz) [inline samples/optimizely-audiences/index.html]: `/>`

File: samples/optimizely-audiences/index.html [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  <link href="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.6-rc.0/css/select2.min.css" rel="stylesheet" />
```
Comment (jelz) [inline samples/optimizely-audiences/index.html]: Be consistent, put `rel` first

File: samples/optimizely-audiences/index.html [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    <div class="cf-form-hint">
+      Please select the target audience for this content.
+    </div>
+    <div class="optimizely-logo"></div>
+  </div>
+
+
+
+  <script>
+    var cfExt = window.contentfulExtension || window.contentfulWidget;
```
Comment (jelz) [inline samples/optimizely-audiences/index.html]: Drop `contentfulWidget`, we shouldn't support prehistoric SDK versions.

It's used once, so just do `window.contentfulExtension.init(...` below.

File: samples/optimizely-audiences/index.html [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        beforeSend: function (request) {
+          // Read-only api token for Optimizely test account. @TODO(floelhoeffel): Move to configuration.
+          request.setRequestHeader('Authorization', 'Bearer 2:5pj23ZOmd4X_pTPR50P_gwl3SjBguvmJ8o2gaqDj0DRg842OFUGc');
+        },
+        url: 'https://api.optimizely.com/v2/audiences?project_id=10467324025',
+        success: function (response) {
+
+          // Filter archived audiences.
+          var audiences = response.filter(function (item) {
+            return !item.archived
```
Comment (jelz) [inline samples/optimizely-audiences/index.html]: end with `;`

File: samples/optimizely-audiences/index.html [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+        errorEl.appendChild(document.createTextNode(msg));
+        document.querySelector('.cf-form-field').style.display = 'none';
+      }
+
+      function createDropDown(data) {
+
+        // Map structure of data so it can seed the dropdown
+        var dropdownData = data.map(function (data) {
+          return { 'id': data.id, 'text': data.name };
```
Comment (jelz) [inline samples/optimizely-audiences/index.html]: no need to wrap keys with apostrophes, `{ id: data.id, ... }` is fine

File: samples/optimizely-audiences/index.html [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        });
+
+        var selectEl = document.createElement('select');
+        selectEl.id = 'audiences-select'
+        selectEl.setAttribute('multiple', 'multiple')
```
Comment (jelz) [inline samples/optimizely-audiences/index.html]: Missing `;`

File: samples/optimizely-audiences/index.html [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+        document.querySelector('.select-container').append(selectEl);
+        var $select2 = $('#audiences-select').select2({
+          placeholder: 'Select audiences',
+          data: dropdownData
+        })
+
+        $select2.on('change', function (e) {
+          var fieldData = [];
+          $select2.select2('data').forEach(function (item) {
```
Comment (jelz) [inline samples/optimizely-audiences/index.html]: You can `map`:

```js
var fieldData = $select2.select2('data').map(function (item) {
  return item.id;
})
```

File: samples/optimizely-audiences/index.html [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+            fieldData.push(item.id);
+          })
+          api.field.setValue(fieldData)
+        });
+
+        var cleanupHandler = api.field.onValueChanged(function () {
```
Comment (jelz) [inline samples/optimizely-audiences/index.html]: `cleanupHandler` seems to be unused.

File: samples/optimizely-audiences/index.html [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+          $select2.val(api.field.getValue());
+          $select2.trigger('change');
+        })
+      }
+    });
+
+  </script>
+</body>
+
+</html>
```
Comment (jelz) [inline samples/optimizely-audiences/index.html]: End files with an empty line.

File: samples/optimizely-audiences/index.html (hunk 1 above)
Comment (floelhoeffel) [reply samples/optimizely-audiences/index.html]: done

File: samples/optimizely-audiences/index.html (hunk 2 above)
Comment (floelhoeffel) [reply samples/optimizely-audiences/index.html]: done

File: samples/optimizely-audiences/index.html (hunk 3 above)
Comment (floelhoeffel) [reply samples/optimizely-audiences/index.html]: done

File: samples/optimizely-audiences/index.html (hunk 4 above)
Comment (floelhoeffel) [reply samples/optimizely-audiences/index.html]: done

File: samples/optimizely-audiences/index.html (hunk 5 above)
Comment (floelhoeffel) [reply samples/optimizely-audiences/index.html]: done

File: samples/optimizely-audiences/index.html (hunk 6 above)
Comment (floelhoeffel) [reply samples/optimizely-audiences/index.html]: done in multiple following cases

File: samples/optimizely-audiences/index.html (hunk 7 above)
Comment (floelhoeffel) [reply samples/optimizely-audiences/index.html]: done

File: samples/optimizely-audiences/index.html (hunk 8 above)
Comment (floelhoeffel) [reply samples/optimizely-audiences/index.html]: I changed so now it uses the same approach from the vanilla template to properly cleanup

File: samples/optimizely-audiences/index.html (hunk 9 above)
Comment (floelhoeffel) [reply samples/optimizely-audiences/index.html]: done

File: samples/optimizely-audiences/index.html [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+
+        detachValueChangeHandler = api.field.onValueChanged(function () {
+          $select2.val(api.field.getValue());
+          $select2.trigger('change');
+        })
+      }
+      // Event handler for window unload.
+      function unloadHandler() {
+        window.removeEventListener('onbeforeunload', unloadHandler);
+        if (cleanupHandler) {
```
Comment (jelz) [inline samples/optimizely-audiences/index.html]: Should be `detachValueChangeHandler`

Comment (floelhoeffel) [reply samples/optimizely-audiences/index.html]: done

**Pull request merged**